## Case Study 1

### 1. You were recently hired as a business analyst in a top sports company. The senior management team has asked you to come up with metrics with which they can gauge which team will win the upcoming La Liga cup (Football tournament). The given data set contains information on all the teams that have so far participated in all the past tournaments. It has data about how many goals each team scored, conceded; how many times they came within the first 6 positions, how many seasons they have qualified, their best position in the past, etc. You are required to do the following:

###### Before doing any analysis it would be a good idea to check for any hypens or other symbols the data set and make appropriate replacements to make sure you can perform arithmetic operations on the data. Prepare a short report to answer the following questions:


In [116]:
import pandas            as pd
import numpy             as np
import matplotlib.pyplot as plt
import seaborn           as sns
import scipy             as stats
import scipy.stats       as stats

In [117]:
df_ll=pd.read_csv('Laliga.csv')
df_ll.head()

Unnamed: 0       Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4 Unnamed: 5  \
0        Pos             Team    Seasons     Points  GamesPlayed   GamesWon   
1          1      Real Madrid         86       4385         2762       1647   
2          2        Barcelona         86       4262         2762       1581   
3          3  Atletico Madrid         80       3442         2614       1241   
4          4         Valencia         82       3386         2664       1187   

   Unnamed: 6 Unnamed: 7 Unnamed: 8    Unnamed: 9 Unnamed: 10 Unnamed: 11  \
0  GamesDrawn  GamesLost   GoalsFor  GoalsAgainst    Champion   Runner-up   
1         552        563       5947          3140          33          23   
2         573        608       5900          3114          25          25   
3         598        775       4534          3309          10           8   
4         616        861       4398          3469           6           6   

  Unnamed: 12 Unnamed: 13 Unnamed: 14 Unnamed: 15 Unnamed: 16 Unnamed: 17  \
0       Third      Fourth       Fifth       Sixth           T       Debut   
1           8           8           3           4          79        1929   
2          12          12           4           6          83        1929   
3          16           9           7           6          56        1929   
4          10          11          10           7          50     1931-32   

     Unnamed: 18   Unnamed: 19  
0  Since/LastApp  BestPosition  
1           1929             1  
2           1929             1  
3        2002-03             1  
4        1987-88             1

In [118]:
df_ll.columns= df_ll.iloc[0]

In [119]:
df_ll=df_ll.iloc[1:61]
df_ll.head()

0 Pos             Team Seasons Points GamesPlayed GamesWon GamesDrawn  \
1   1      Real Madrid      86   4385        2762     1647        552   
2   2        Barcelona      86   4262        2762     1581        573   
3   3  Atletico Madrid      80   3442        2614     1241        598   
4   4         Valencia      82   3386        2664     1187        616   
5   5  Athletic Bilbao      86   3368        2762     1209        633   

0 GamesLost GoalsFor GoalsAgainst Champion Runner-up Third Fourth Fifth Sixth  \
1       563     5947         3140       33        23     8      8     3     4   
2       608     5900         3114       25        25    12     12     4     6   
3       775     4534         3309       10         8    16      9     7     6   
4       861     4398         3469        6         6    10     11    10     7   
5       920     4631         3700        8         7    10      5     8    10   

0   T    Debut Since/LastApp BestPosition  
1  79     1929          1929            1  
2  83     1929          1929            1  
3  56     1929       2002-03            1  
4  50  1931-32       1987-88            1  
5  49     1929          1929            1

In [120]:
for i in df_ll['Debut'].index:
    x=df_ll['Debut'][i]
    year=x[0:4]
    df_ll['Debut'][i]=int(year)

In [121]:
for i in df_ll['Since/LastApp'].index:
    x=df_ll['Since/LastApp'][i]
    if '-' in x:
        year=x[0:4]
        df_ll['Since/LastApp'][i]=int(year)+1
    else:
        year=x[0:4]
        df_ll['Since/LastApp'][i]=int(year)

#### 1.	Which are the teams which started playing between 1930-1980?

In [122]:
print(df_ll[(df_ll['Debut']>=1930) & (df_ll['Debut']<=1980)]['Team'])

4                Valencia
6                 Sevilla
9                Zaragoza
10             Real Betis
11    Deportivo La Coruna
12             Celta Vigo
13             Valladolid
15         Sporting Gijon
16                Osasuna
17                 Malaga
18                 Oviedo
19               Mallorca
20             Las Palmas
22                Granada
23         Rayo Vallecano
24                  Elche
26               Hercules
27               Tenerife
28                 Murcia
29                 Alaves
30                Levante
31              Salamanca
32               Sabadell
33                  Cadiz
35              Castellon
38                Cordoba
40             Recreativo
41              Burgos CF
42             Pontevedra
47              Gimnastic
50               Alcoyano
51                   Jaen
53             AD Almeria
55                 Lleida
58                 Condal
59        Atletico Tetuan
60       Cultural Leonesa
Name: Team, dtype: object


#### 2.	Which are the top 5 teams in terms of points?

In [123]:
df_ll=df_ll.astype({'Points':int})

In [124]:
df_ll.sort_values(by='Points',ascending=False)['Team'].head(5)

1        Real Madrid
2          Barcelona
3    Atletico Madrid
4           Valencia
5    Athletic Bilbao
Name: Team, dtype: object

#### 3.	What is the distribution of the winning percentage for all teams? Which teams are in the top 5 in terms of winning percentage? 

##### (Winning percentage= (GamesWon / GamesPlayed)*100)

In [125]:
df_ll=df_ll.astype({'GamesWon':int,'GamesPlayed':int})

In [126]:
df_ll['Winning_%']=(df_ll['GamesWon']/df_ll['GamesPlayed'])*100

In [127]:
df_ll.sort_values(by='Winning_%',ascending=False)['Team'].head(5)

1        Real Madrid
2          Barcelona
3    Atletico Madrid
4           Valencia
5    Athletic Bilbao
Name: Team, dtype: object

#### 4.	Is there a significant difference in the winning percentage for teams which have attained a best position between 1-3 and those teams which have had a best position between 4-7

In [128]:
df_ll=df_ll.astype({'BestPosition':int})

In [129]:
pos_1_3=list(df_ll[(df_ll['BestPosition']>=1) & (df_ll['BestPosition']<=3)]['Winning_%'])
pos_4_7=list(df_ll[(df_ll['BestPosition']>=4) & (df_ll['BestPosition']<=7)]['Winning_%'])

In [130]:
stats.ttest_ind(pos_1_3,pos_4_7)

Ttest_indResult(statistic=4.992715339447283, pvalue=1.5362101870160483e-05)

##### As the p-value is less than 0.05 we can conclude that there is a significant difference in the winning percentage for teams which have attained a best position between 1-3 and those teams which have had a best position between 4-7

## case study 2

### A study was done to measure the blood pressure of 60 year old women with glaucoma. A random sample of 200 60-year old women with glaucoma was chosen. The mean of the systolic blood pressure in the sample was 140 mm Hg and the standard deviation was 25 mm Hg.

In [131]:
n=200
xbar=140
sigma=25

#### a)	Calculate the estimated standard error of the sample mean? What does the standard error indicate?

In [132]:
se=sigma/np.sqrt(n)
print('standard error:',se)

standard error: 1.7677669529663687


#### b)	Estimate a 95% confidence interval for the true mean blood pressure for all 60-year old women with glaucoma.

In [133]:
x1,x2=stats.norm.interval(0.95,loc=xbar,scale=se)
print(x1,x2)

136.5352404391258 143.4647595608742


##### systolic pressure range for all 60-year old women with glaucoma lies in betwee 136.53 mm Hg and 143.46 mm Hg

#### c)	Assume that instead of 200, a random sample of only 100 60-year old women with glaucoma was chosen. The sample mean and standard deviation estimates are the same as those in the original study. What is the estimated 95% confidence interval for the true mean blood pressure?

In [134]:
n=100
xbar=140
sigma=25
se=sigma/np.sqrt(n)
y1,y2=stats.norm.interval(0.95,loc=xbar,scale=se)
print(y1,y2)
print(se)

135.10009003864985 144.89990996135015
2.5


##### systolic pressure range for all 60-year old women with glaucoma lies in betwee 135.1 mm Hg and 144.89 mm Hg

#### d)	Which of the two above intervals are wider?

In [135]:
print('interval1 width:',x2-x1)
print('interval2 width:',y2-y1)

interval1 width: 6.929519121748399
interval2 width: 9.799819922700294


##### interval of 100 samples is wider

#### e)	Explain in non-technical terms why the estimated standard error of a sample mean tends to decrease with an increase in sample size

##### Standard error decreases when sample size increases – as the sample size gets closer to the true size of the population, the sample means cluster more and more around the true population mean.

## case study 3

### 3. Par Inc., is a major manufacturer of golf equipment. Management believes that Par’s market share could be increased with the introduction of a cut-resistant, longer-lasting golf ball. Therefore, the research group at Par has been investigating a new golf ball coating designed to resist cuts and provide a more durable ball. The tests with the coating have been promising.

### One of the researchers voiced concern about the effect of the new coating on driving distances. Par would like the new cut-resistant ball to offer driving distances comparable to those of the current-model golf ball. To compare the driving distances for the two balls, 40 balls of both the new and current models were subjected to distance tests. The testing was performed with a mechanical hitting machine so that any difference between the mean distances for the two models could be attributed to a difference in the design. The results of the tests, with distances measured to the nearest yard, are contained in   the data set “Golf”. 

In [136]:
df_golf= pd.read_csv('Golf.csv')
df_golf.head()

Current  New
0      264  277
1      261  269
2      267  263
3      272  266
4      258  262

#### 1.	Formulate and present the rationale for a hypothesis test that Par could use to compare the driving distances of the current and new golf balls.

##### H0: The driving distance of the new ball is NOT significantly greater than that of old ball.
##### HA: The driving distance of the new ball is significantly greater than that of old ball.

#### 2.	Analyze the data to provide the hypothesis testing conclusion. What is the p-value for your test? What is your recommendation for Par Inc.?

In [137]:
stats.ttest_rel(df_golf['Current'],df_golf['New'])

Ttest_relResult(statistic=1.2769699827911767, pvalue=0.20916361823147053)

##### The p-value is greater than 0.05 ,so we fail to reject Null hypothesis.
##### Therefore we can conclude that there is no significant difference between the driving distance of the new ball than that of the old ball.

#### 3.	What is the 95% confidence interval for the population mean of each model, and what is the 95% confidence interval for the difference between the means of the two population?

##### model 1 current

In [138]:
n=len(df_golf['Current'])
xbar1=np.mean(df_golf['Current'])
sigma1=np.std(df_golf['Current'])
se1=sigma1/np.sqrt(n)
print('SE current:',se1)
x1,x2=stats.norm.interval(0.95,loc=xbar1,scale=se1)
print('95% confidence interval of driving distance of the current ball',x1,x2)

SE current: 1.3665593199711457
95% confidence interval of driving distance of the current ball 267.596592950119 272.953407049881


##### model 1 new

In [139]:
n=len(df_golf['New'])
xbar2=np.mean(df_golf['New'])
sigma2=np.std(df_golf['New'])
se2=sigma2/np.sqrt(n)
print('SE current:',se1)
y1,y2=stats.norm.interval(0.95,loc=xbar2,scale=se2)
print('95% confidence interval of driving distance of the new ball',y1,y2)

SE current: 1.3665593199711457
95% confidence interval of driving distance of the new ball 264.4715543699107 270.5284456300893


In [142]:
n=df_golf.shape[0]
mean_diff=np.mean(df_golf['Current']-df_golf['New'])
std_diff=np.std(df_golf['Current']-df_golf['New'])
SE=std_diff/np.sqrt(n)
CI_diff=stats.t.interval(0.95,df=n-1,loc=mean_diff,scale=SE)
print('95% CI of diff between two means of two population:',CI_diff)

95% CI of diff between two means of two population: (-1.565243844488808, 7.115243844488807)
